### PR TITLE
Golden autoRIFT test

### PR DESCRIPTION
**Pull request by jhkennedy**
This provides a golden comparison for autoRIFT jobs

* ~~requires the product naming changes in ASFHyP3/hyp3-autorift#23~~ merged!

### TODOs:

- [x] test `hyp3_testing.compare`
- [x] manual GitHub action for `test_autorift`

### CODE REVIEW

File: .github/workflows/golden_autorift.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Golden RTC
```
Comment (asjohnston-asf) [inline .github/workflows/golden_autorift.yml]: `Golden autoRIFT`

File: conda-env.yml [hunk 2]
```diff
@@ -15,7 +15,9 @@ dependencies:
   - pytest-dependency
   - pytest-timeout
   - requests
+  - rasterio
   - wheel
+  - xarray
```
Comment (asjohnston-asf) [inline conda-env.yml:21]: `numpy` was also added in `setup.py`, should it be here as well?

File: tests/test_compare.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    ref_file.write_text('hello')
+
+    sec_file = tmp_path / 'sec.txt'
+    sec_file.write_text('hello')
+
+    compare.bit_for_bit(ref_file, sec_file)
+
+    with pytest.raises(compare.ComparisonFailure):
+        sec_file.write_text('Goodbye!')
+        compare.bit_for_bit(ref_file, sec_file)
```
Comment (asjohnston-asf) [inline tests/test_compare.py:24]: should also test the case were the two files are the same size, but differ

**Comment by jhkennedy** [comment]
I ran this entirely through locally (submit, wait, download, and compare) with the new autoRIFT deployment and everything but the date_created attributes are the same

```
E           hyp3_testing.compare.ComparisonFailure: 4 of 4 products are different!
E           
E           --------------------------------------------------------------------------------
E           S1AA_20151120T080202_20151214T080202_HHP024_VEL240_A_{4E88,EEC0}
E           --------------------------------------------------------------------------------
E           Reference and secondary Dataset objects are not identical
E           
E           
E           Differing attributes:
E           
E           R   date_created: 14-Oct-2020 03:36:29
E           S   date_created: 14-Oct-2020 03:29:00
E           
E           --------------------------------------------------------------------------------
E           S1AA_20150828T162412_20150909T162413_HHP013_VEL240_A_{6F8F,6249}
E           --------------------------------------------------------------------------------
E           Reference and secondary Dataset objects are not identical
E           
E           
E           Differing attributes:
E           
E           R   date_created: 14-Oct-2020 03:08:00
E           S   date_created: 14-Oct-2020 02:59:25
E           
E           --------------------------------------------------------------------------------
E           S1BB_20180809T204617_20180821T204618_HHP013_VEL240_A_{A259,6962}
E           --------------------------------------------------------------------------------
E           Reference and secondary Dataset objects are not identical
E           
E           
E           Differing attributes:
E           
E           R   date_created: 14-Oct-2020 03:41:39
E           S   date_created: 14-Oct-2020 03:31:08
E           
E           --------------------------------------------------------------------------------
E           S1AB_20170221T204710_20170227T204628_HHP006_VEL240_A_{DF8D,F036}
E           --------------------------------------------------------------------------------
E           Reference and secondary Dataset objects are not identical
E           
E           
E           Differing attributes:
E           
E           R   date_created: 14-Oct-2020 03:48:58
E           S   date_created: 14-Oct-2020 03:44:28
```

File: hyp3_testing/helpers.py [hunk 4]
```diff
@@ -66,9 +66,10 @@ def get_download_urls(jobs: List[dict]) -> List[str]:
 def download_products(jobs: List[dict], directory: Path):
     urls = get_download_urls(jobs)
     for url in urls:
-        zip_file = download_file(url, directory=str(directory))
-        with ZipFile(zip_file) as zip_:
-            zip_.extractall(path=directory)
+        product_file = download_file(url, directory=str(directory))
+        if url.endswith('.zip'):
```
Comment (asjohnston-asf) [inline hyp3_testing/helpers.py]: if `product_file.endswith('.zip')`?  same thing but it feels a little better.  Or `if zipfile.is_zipfile(product_file):`

File: tests/test_autorift.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+from glob import glob
+from pathlib import Path
+
+import pytest
+import xarray as xr
+
+from hyp3_testing import API_TEST_URL, API_URL
+from hyp3_testing import helpers
+from hyp3_testing.compare import ComparisonFailure
+from hyp3_testing.compare import bit_for_bit, compare_cf_spatial_reference, values_are_close
```
Comment (asjohnston-asf) [inline tests/test_autorift.py]: two `compare` imports?  Should these be one line?

File: hyp3_testing/helpers.py (hunk 4 above)
Comment (jhkennedy) [reply hyp3_testing/helpers.py]: I like the `is_zipfile` approach

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**